### ISSUE TITLE
Failed to load board file pciid [17cb:1103] subsystem id [105b:e0b8]

### ISSUE THREAD

**Issue by KaiChuan-Hsieh** [kvalo/ath11k-firmware#10]
Dec 21 15:02:13 ubuntu kernel: ath11k_pci 0000:71:00.0: failed to fetch board data for bus=pci,vendor=17cb,device=1103,subsystem-vendor=105b,subsystem-device=e0b8,qmi-chip-id=2,qmi-board-id=255 from ath11k/WCN6855/hw2.0/board-2.bin
Dec 21 15:02:13 ubuntu kernel: ath11k_pci 0000:71:00.0: failed to fetch board-2.bin or board.bin from WCN6855/hw2.0
Dec 21 15:02:13 ubuntu kernel: ath11k_pci 0000:71:00.0: qmi failed to fetch board file: -2
Dec 21 15:02:13 ubuntu kernel: ath11k_pci 0000:71:00.0: failed to load board data file: -2
[load-board-file-failed.log](https://github.com/kvalo/ath11k-firmware/files/7750735/load-board-file-failed.log)

**Comment by bihagkashikar** [comment]
same error for me too - have tried loading the bin files from QC_IMAGE_VERSION_STRING=WLAN.HSP.1.1-**03003**-QCAHSPSWPL_V1_V2_SILICONZ_LITE-2 and QC_IMAGE_VERSION_STRING=WLAN.HSP.1.1-**02892.1**-QCAHSPSWPL_V1_V2_SILICONZ_LITE-1

[   14.328273] ath11k_pci 0000:03:00.0: chip_id 0x2 chip_family 0xb board_id 0xff soc_id 0x400c0200
[   14.328277] ath11k_pci 0000:03:00.0: fw_version 0x110c0b4c fw_build_timestamp 2021-11-24 04:24 fw_build_id QC_IMAGE_VERSION_STRING=WLAN.HSP.1.1-02892.1-QCAHSPSWPL_V1_V2_SILICONZ_LITE-1
[   14.329460] ath11k_pci 0000:03:00.0: failed to fetch board data for bus=pci,qmi-chip-id=2,qmi-board-id=255 from ath11k/WCN6855/hw2.0/board-2.bin
[   14.329485] ath11k_pci 0000:03:00.0: failed to fetch board-2.bin or board.bin from WCN6855/hw2.0
[   14.329486] ath11k_pci 0000:03:00.0: failed to fetch board file: -2
[   14.329487] ath11k_pci 0000:03:00.0: failed to load board data file: -2

hoping this will be resolved soon.

**Comment by ZeekoZhu** [comment]
Maybe you can try these files https://github.com/kvalo/ath11k-firmware/tree/0e2a81c430cad7b7a2afaf9a7aa4cecf958227c8/WCN6855/hw2.0/1.1

**Comment by bihagkashikar** [comment]
thanks, I have the above one's too and one's from below - 

https://chromium.googlesource.com/chromiumos/third_party/linux-firmware/+/5e5cab716921f79d79d8518d3718669d18fc2972/ath11k/WCN6855/hw2.0 

and now the error is not appearing in the log, however, i see this, but no wifi option appears so not sure what to make of this. don't think wifi lan is enabled with these driver files

an 15 15:38:27 pop-os kernel: ath11k_pci 0000:03:00.0: BAR 0: assigned [mem 0xfd000000-0xfd1fffff 64bit]
Jan 15 15:38:27 pop-os kernel: ath11k_pci 0000:03:00.0: enabling device (0000 -> 0002)
Jan 15 15:38:27 pop-os kernel: ath11k_pci 0000:03:00.0: wcn6855 hw2.0

**Comment by ZeekoZhu** [comment]
@bihagkashikar  These files (https://chromium.googlesource.com/chromiumos/third_party/linux-firmware/+/a4577b90a7edc50f2098c7b4f5b87a0688bfe10d/ath11k/WCN6855/hw2.0) works for me (kernel 5.15)

**Comment by bihagkashikar** [comment]
@ZeekoZhu  thanks again. 

I copied the files from the link you provided and I see the same log, and yes, I am on 5.15 kernel too.

Did you carry out any further steps after copying those files?

an 15 15:38:27 pop-os kernel: ath11k_pci 0000:03:00.0: BAR 0: assigned [mem 0xfd000000-0xfd1fffff 64bit]
Jan 15 15:38:27 pop-os kernel: ath11k_pci 0000:03:00.0: enabling device (0000 -> 0002)
Jan 15 15:38:27 pop-os kernel: ath11k_pci 0000:03:00.0: wcn6855 hw2.0

**Comment by ZeekoZhu** [comment]
@bihagkashikar After copied these files, I added a file to load ath11k_pci module automaticly:

*/etc/modules-load.d/ath11k.conf*
```
ath11k_pci
```

And here is my log:
```
❯ journalctl --since today | grep ath11k
Jan 18 08:56:55 zhutao-hp-arch kernel: ath11k_pci 0000:01:00.0: BAR 0: assigned [mem 0xfb000000-0xfb1fffff 64bit]
Jan 18 08:56:55 zhutao-hp-arch kernel: ath11k_pci 0000:01:00.0: enabling device (0000 -> 0002)
Jan 18 08:56:55 zhutao-hp-arch kernel: ath11k_pci 0000:01:00.0: wcn6855 hw2.0
Jan 18 08:56:55 zhutao-hp-arch systemd-modules-load[337]: Inserted module 'ath11k_pci'
Jan 18 08:56:57 zhutao-hp-arch kernel: Modules linked in: qrtr_mhi snd_hda_core snd_hwdep kvm_amd snd_pcm snd_timer snd_pci_acp5x hid_multitouch(+) gpu_sched hp_wmi(+) drm_ttm_helper sparse_keymap vfat pcc_cpufreq(-) joydev snd snd_rn_pci_acp3x fjes(-) platform_profile fat wmi_bmof mousedev snd_pci_acp3x kvm soundcore ttm ccp irqbypass crct10dif_pclmul crc32_pclmul ghash_clmulni_intel aesni_intel tpm_crb crypto_simd uvcvideo cryptd videobuf2_vmalloc rapl videobuf2_memops ucsi_acpi videobuf2_v4l2 btusb sp5100_tco typec_ucsi videobuf2_common btrtl pcspkr i2c_piix4 k10temp typec btbcm tpm_tis roles wmi btintel tpm_tis_core tpm bluetooth wireless_hotkey rng_core i2c_hid_acpi video pinctrl_amd i2c_hid cm32181 ecdh_generic industrialio acpi_cpufreq amd_pmc mac_hid crc16 usbhid vmmon(OE) vmw_vmci v4l2loopback(OE) videodev mc pkcs8_key_parser crypto_user qrtr ns ath11k_pci ath11k qmi_helpers mac80211 libarc4 cfg80211 rfkill mhi fuse bpf_preload ip_tables x_tables serio_raw atkbd libps2 xhci_pci i8042 xhci_pci_renesas
Jan 18 08:56:57 zhutao-hp-arch kernel:  ath11k_qmi_msg_mem_request_cb+0x1b6/0x310 [ath11k 0241a10f9a13cf393366ee1621fb4dfdd2548406]
Jan 18 08:56:57 zhutao-hp-arch kernel: ath11k_pci 0000:01:00.0: chip_id 0x0 chip_family 0xb board_id 0xff soc_id 0x400c0200
Jan 18 08:56:57 zhutao-hp-arch kernel: ath11k_pci 0000:01:00.0: fw_version 0x110406b8 fw_build_timestamp 2021-01-08 06:14 fw_build_id QC_IMAGE_VERSION_STRING=WLAN.HSP.1.1-01720.1-QCAHSPSWPL_V1_V2_SILICONZ_LITE-1
Jan 18 08:56:57 zhutao-hp-arch kernel: ath11k_pci 0000:01:00.0: failed to fetch board data for bus=pci,qmi-chip-id=0,qmi-board-id=255 from ath11k/WCN6855/hw2.0/board-2.bin
Jan 18 08:56:58 zhutao-hp-arch NetworkManager[530]: <info>  [1642467418.3137] rfkill1: found Wi-Fi radio killswitch (at /sys/devices/pci0000:00/0000:00:01.3/0000:01:00.0/ieee80211/phy0/rfkill1) (driver ath11k_pci)
```

**Comment by bihagkashikar** [comment]
@ZeekoZhu thanks again. 

I added ath11k.conf and even then I don't see anything in log after ath11k_pci 0000:01:00.0: wcn6855 hw2.0 this line.

Could you please let me know output of  this command _modinfo ath11k_pci_

For me, the output of modinfo shows firmware path to QCA6390 instead of WCN6855. Not sure how to update the path to point to WCN6855.


filename:       /lib/modules/5.15.11-76051511-generic/kernel/drivers/net/wireless/ath/ath11k/ath11k_pci.ko
firmware:       ath11k/QCA6390/hw2.0/m3.bin
firmware:       ath11k/QCA6390/hw2.0/amss.bin
firmware:       ath11k/QCA6390/hw2.0/board-2.bin
license:        Dual BSD/GPL
description:    Driver support for Qualcomm Technologies 802.11ax WLAN PCIe devices
srcversion:     DB13C105C3498A7DA4F3FE4
alias:          pci:v000017CBd00001104sv*sd*bc*sc*i*
alias:          pci:v000017CBd00001103sv*sd*bc*sc*i*
alias:          pci:v000017CBd00001101sv*sd*bc*sc*i*
depends:        mhi,ath11k
retpoline:      Y
intree:         Y
name:           ath11k_pci
vermagic:       5.15.11-76051511-generic SMP mod_unload modversions 
sig_id:         PKCS#7
signer:         Build time autogenerated kernel key
sig_key:        78:9B:13:21:9E:70:A4:D5:B6:09:62:0E:35:BB:4F:2A:65:42:F8:95
sig_hashalgo:   sha512

**Comment by ZeekoZhu** [comment]
@bihagkashikar Maybe you can try to load the `ath11k_pci` module manually, then check if it has been loaded.

```shell
$ sudo modprobe ath11k_pci
$ lsmod  | grep ath11k_pci
ath11k_pci             32768  0
ath11k                507904  1 ath11k_pci
mhi                   102400  2 ath11k_pci,qrtr_mhi
```

> For me, the output of modinfo shows firmware path to QCA6390 instead of WCN6855. Not sure how to update the path to point to WCN6855.

Sorry, I don't know how to change the path to WCN6855, my `modinfo` output is the same as yours.

**Comment by kvalo** [comment]
For WCN6855 I recommend trying just released v5.17-rc1.

Also please file bugs to bugzilla:

https://wireless.wiki.kernel.org/en/users/drivers/ath11k/bugreport
